Meta/gn/build/extract_archive_contents.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+r"""Extracts files from an archive for use in the build
+
+It's intended to be used for files that are cached between runs.
+
+"""
+
+import argparse
+import pathlib
+import tarfile
+import zipfile
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser(
+                 epilog=__doc__,
+                 formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('archive', help='input archive')
+    parser.add_argument('paths', nargs='*', help='paths to extract from the archive')
+    parser.add_argument('-d', "--destination", required=True,
+                        help='directory to write the extracted file to')
+    args = parser.parse_args()
+
+    archive = pathlib.Path(args.archive)
+    destination = pathlib.Path(args.destination)
+
+    def extract_member(file, path):
+        destination_path = destination / path
+        if destination_path.exists():
+            return
+        destination_path.parent.mkdir(parents=True, exist_ok=True)
+        if isinstance(file, tarfile.TarFile):
+            with file.extractfile(path) as member:
+                destination_path.write_text(member.read().decode('utf-8'))
+        else:
+            assert isinstance(file, zipfile.ZipFile)
+            try:
+                with file.open(path) as member:
+                    destination_path.write_text(member.read().decode('utf-8'))
+                return
+            except KeyError:
+                pass
+            # Assume that this is actually a *directory* rather than a filename
+            # FIXME: Do this smarter... Can we convince GN to let us put directorie
```

Comment:
You can write as many files as you want without telling the build system about it. You just can't have build system dependency chains for those files.

Reply:
Hm sounds like in this case the "answer" is to pass the args differently for the CLDR invocation that wants to extract (8k!) files into ~8 directories rather than N < 10 individual files.